Packages/Sequence-Unity/Sequence/SequenceSDK/Ethereum/ChainId.cs

```diff
@@ -23,5 +23,20 @@ public static string GetChainId(this Chain chain)
         {
             return ChainDictionaries.ChainIdOf[chain];
         }
+
     }
+
+    public static class NativeTokenAddress
+    {
+        public static string GetNativeTokenAddress(Chain chain)
+        {
+            return ChainDictionaries.NativeTokenAddressOf.TryGetValue(chain, out var address) ? address : null;
```

Comment:
This should only ever fail if we've made a mistake when maintaining the codebase (forgetting to add a chain to the list). Returning null could lead to unexpected and hard to diagnose errors for integrators. Instead, we should throw an exception directly so if this does come up the error is easier to diagnose.

In addition, we should add a test suite to check that all of these are set - look at ChainIndexerTests.TestCreateChainIndexerForChain

Reply:
updated

Reply:
An easy way to test this (as well as an important chunk of logic) would be to test that we can create a QR code link to transfer a random amount of native tokens on each network